ogre2/src/Ogre2GpuRays.cc

```diff
@@ -589,6 +794,24 @@ void Ogre2GpuRays::Setup1stPass()
     this->dataPtr->ogreCompositorWorkspace1st[i] =
         ogreCompMgr->addWorkspace(this->scene->OgreSceneManager(),
         rt, this->dataPtr->cubeCam[i], wsDefName, false);
+
+    // add laser retro material switcher to render target listener
+    // so we can switch to use laser retro material when the camera is being udpated
```

Comment:
```suggestion
    // so we can switch to use laser retro material when the camera is being updated
```

I think there is also a line length problem here.